src/basic/format-util.c

```diff
@@ -5,21 +5,43 @@
 #include "stdio-util.h"
 
 assert_cc(DECIMAL_STR_MAX(int) + 1 <= IF_NAMESIZE + 1);
-char *format_ifname_full(int ifindex, char buf[static IF_NAMESIZE + 1], FormatIfnameFlag flag) {
+int format_ifname_full(int ifindex, FormatIfnameFlag flag, char buf[static IF_NAMESIZE + 1]) {
         /* Buffer is always cleared */
         memzero(buf, IF_NAMESIZE + 1);
```

Comment:
hmm, so this is actually weird. Our coding style mandates we don't touch output params on failure... I'd prefer if we'd not zero this out here on failure. it's just too surprising if this function behaves differently from all others...

Reply:
(it might be OK if we had a flag that enables this behaviour, i.e. FORMAT_IFNAME_EMPTY_ON_FAILURE or so. but in that case we should eat up all errors, and return success)

Reply:
Addressed.